plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy

```diff
@@ -194,14 +194,52 @@ class SchemaValidator extends PluginExtensionPoint {
         return channel
     }
 
+
+    //
+    // Initialise expected params if not present
+    //
+    Map initialiseExpectedParams(Map params) {
+        if( !params.containsKey("validationFailUnrecognisedParams") ) {
+            params.validationFailUnrecognisedParams = false
+        }
+        if( !params.containsKey("validationLenientMode") ) {
+            params.validationLenientMode = false
+        }
+        if( !params.containsKey("monochrome_logs") ) {
+            params.monochrome_logs = false
+        }
+        if( !params.containsKey("help") ) {
+            params.help = false
+        }
+        if( !params.containsKey("validationShowHiddenParams") ) {
+            params.validationShowHiddenParams = false
+        }
+        if( !params.containsKey("schema_ignore_params") ) {
+            params.schema_ignore_params = false
+        }
+
```

Comment:
Can you add `immutable_meta` with a default of true here too?

Reply:
Should we add it in camelCase like the others? I would suggest `validationImmutableMeta`
(I refactored `schema_ignore_params` too in a different PR)